coverage/PENDING_REVIEW/intel/0490-COVERAGE-nir-split-vars-1199/cov-uninitialized-values-passed-to-function-never-executed.amber

```diff
@@ -0,0 +1,265 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific NIR code paths
+
+# The test passes because the shader always writes red.
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+#
+# #define _int_0 _GLF_uniform_int_values[0]
+# #define _int_1 _GLF_uniform_int_values[1]
+# #define _int_10 _GLF_uniform_int_values[2]
+#
+# precision highp float;
+# precision highp int;
+#
+# // Contents of _GLF_uniform_int_values: [0, 1, 10]
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     int _GLF_uniform_int_values[3];
+# };
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# struct S
+# {
+#     int data;
+# };
+#
+# void func(inout S s, int x)
+# {
+#     // Always false.
+#     if (_int_1 == _int_0)
+#     {
+#         return;
+#     }
+#
+#     s.data = x;
+# }
+#
+# void main()
+# {
+#     // These variables need to be uninitialized to hit the coverage point.
+#     // However, the uninitialized values are never read.
+#     int index;
+#     S arr[10];
+#
+#     // Always false.
+#     if (_int_1 == _int_0)
+#     {
+#         // Uninitialized value access that is never executed.
+#         func(arr[index], index);
+#     }
+#     else
+#     {
+#         func(arr[_int_0], _int_1);
```

Comment:
This looks invalit do me: arr[0] is read here, but it is uninitialized.

Reply:
Isn't the point that due to func's first argument being inout, and not read, it's OK that arr[0] is uninitialized? Can that be commented?

Can func's argument be made `out` only? That would be even clearer.

Can the comment on the "then" side explicitly comment about `index` being uninitialized? Because `arr` is also uninitialized I initially misunderstood this.

Reply:
Using just _out_ makes the coverage point disappear, but it looks like the array can be initialized. So currently only _index_ is left unitialized, and that gets never read.